detector.go

```diff
@@ -235,12 +204,24 @@ func (bp *buildpackTOML) Detect(c *DetectConfig) detectRun {
 		return detectRun{Code: -1, Err: err}
 	}
 	defer os.RemoveAll(planDir)
+	if err := ensureOwner(planDir, c.UID, c.GID); err != nil {
```

Comment:
Let's chat about these changes to `Builder` and `Detector`. Having these sections of code run as root while writing files and modifying the environment seems error prone and worries me from a security perspective. I'm especially worried about untrusted input from app/buildpack artifacts that could accidentally be executed as root.